cluster-provision/images/vm-image-builder/customize-image.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+set -exuo pipefail
+
+function cleanup() {
+  if [ $? -ne 0 ]; then
+    rm -f "${CUSTOMIZE_IMAGE_PATH}"
+  fi
+
+  rm -rf "${CLOUD_INIT_ISO}"
+  virsh destroy "${DOMAIN_NAME}" || true
+  virsh undefine "${DOMAIN_NAME}" || true
+}
+
+SOURCE_IMAGE_PATH=$1
+OS_VARIANT=$2
+CUSTOMIZE_IMAGE_PATH=$3
+CLOUD_CONFIG_PATH=$4
+
+readonly DOMAIN_NAME="provision-vm"
+readonly CLOUD_INIT_ISO="cloudinit.iso"
+
+trap 'cleanup' EXIT
```

Comment:
there are multi traps, please see the discussion about SIGTERM and SIGINT 
(we will continue discussion on the other threads)

(wont write it on each one, please consider on each trap occurence)